fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/ParquetMergeJob.java

```diff
@@ -0,0 +1,169 @@
+
+    private String outputPath;
+
+    public ParquetMergeJob(String pathPattern, long tableId, EtlJobConfig.EtlIndex indexMeta,
+                           SerializableConfiguration serializableHadoopConf, String outputPath) {
+        this.pathPattern = pathPattern;
+        this.tableId = tableId;
+        this.indexMeta = indexMeta;
+        this.serializableHadoopConf = serializableHadoopConf;
+        this.outputPath = outputPath;
+    }
+
+    @Override
+    public Void apply(Iterator<Map.Entry<String, Integer>> v1) {
+        TaskContext taskContext = TaskContext.get();
+        long taskAttemptId = taskContext.taskAttemptId();
+        Configuration conf = new Configuration(serializableHadoopConf.value());
+        conf.setBoolean("spark.sql.parquet.writeLegacyFormat", false);
+        conf.setBoolean("spark.sql.parquet.int64AsTimestampMillis", false);
+        conf.setBoolean("spark.sql.parquet.int96AsTimestamp", true);
+        conf.setBoolean("spark.sql.parquet.binaryAsString", false);
+        conf.set("spark.sql.parquet.outputTimestampType", "INT96");
+
+        try {
+            FileSystem fs = FileSystem.get(URI.create(outputPath), conf);
+            while (v1.hasNext()) {
+                Map.Entry<String, Integer> divideEntry = v1.next();
+                String bucketKey = divideEntry.getKey();
+                String partitionId = bucketKey.split("_")[0];
+                String bucketId = bucketKey.split("_")[1];
+                int divideN = divideEntry.getValue();
+                String dstPath = String.format(pathPattern, tableId, Integer.parseInt(partitionId), indexMeta.indexId,
+                        Integer.parseInt(bucketId), indexMeta.schemaHash);
+                String tmpPath = dstPath + "." + taskAttemptId;
+                List<Path> fileList = getFilePathList(fs, divideN, dstPath);
+
+                LOG.info("fileList.get(0) = " + fileList.get(0));
+                LOG.info("tmpPath = " + tmpPath);
+
+                HadoopInputFile inputFile = HadoopInputFile.fromPath(fileList.get(0), conf);
+                ParquetFileReader parquetFileReader = ParquetFileReader.open(inputFile);
+                ParquetMetadata metadata = parquetFileReader.getFooter();
+                MessageType schema = metadata.getFileMetaData().getSchema();
+                Group group = null;
+                ParquetReader<Group> parquetReader = null;
+                GroupWriteSupport.setSchema(schema, conf);
+                GroupWriteSupport writeSupport = new GroupWriteSupport();
+                ParquetWriter parquetWriter = new ParquetWriter(new Path(dstPath), writeSupport,
+                        CompressionCodecName.SNAPPY,
+                        256 * 1024 * 1024, 16 * 1024,
+                        1024 * 1024,
+                        true, false,
+                        ParquetProperties.WriterVersion.PARQUET_1_0,
+                        conf);
+                for (Path path : fileList) {
```

Comment:
How to ensure that the files are globally ordered?

Reply:
by calculating a boundary, the data can be split according to the boundary to achieve local order to global order, similar to the sortby in spark.i have two ways to re-partition, but using hash for re-partition is problematic as it does not consider the case of on sorting key having multiple values, I have already removed this method.